entrypoint.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+set -ex
+
+if [[ -z "$*" ]]; then
+  echo "Expecting one or more arguments"
+  exit 1
+fi
+
+#remove spack if exist previously
+if [[   -d "$GITHUB_WORKSPACE"/temp/spack ]]; then
+    rm -rf "$GITHUB_WORKSPACE"/temp
+fi
+
+# install Spack
+git clone https://github.com/spack/spack  "$GITHUB_WORKSPACE"/temp/spack&& \
+    PATH=$PATH:"$GITHUB_WORKSPACE"/temp/spack/bin && \
+    echo "$PATH" && export PATH && \
+    bash "$GITHUB_WORKSPACE"/temp/spack/share/spack/setup-env.sh
+
+if [[ -n "$SPACK_RESET_CONFIG" ]]; then
+  if [[ -z "$SPACK_GIT_REF" ]]; then
+    echo "Please provide the git ref to reset"
```

Comment:
we can mention the variable name so users know what to specify, something like `Expecting SPACK_GIT_REF variable`